examples/clients/Imaging.c

```diff
@@ -0,0 +1,90 @@
+/** @file
+    @brief Implementation
+
+    @date 2015
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com/osvr>
+*/
+
+// Copyright 2015 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/ClientKit/ContextC.h>
+#include <osvr/ClientKit/InterfaceC.h>
+#include <osvr/ClientKit/InterfaceCallbackC.h>
+#include <osvr/ClientKit/ImagingC.h>
+
+// Standard includes
+#include <stdio.h>
+
+unsigned int reportNumber = 0;
+
+void imagingCallback(void *userdata, const OSVR_TimeValue *timestamp,
+                      const OSVR_ImagingReport *report) {
+    // @todo implement a C equivalent to the C++ empty() check here.
+    OSVR_ClientContext* ctx = (OSVR_ClientContext*)userdata;
+    /// The first time, let's print some info.
+    if (reportNumber == 0) {
+        printf("Got first report: image is %d width and %d height.\n",
+        report->state.metadata.width, report->state.metadata.height);
+    }
+    else {
+        printf("Got report number %d\n", reportNumber);
+    }
+
+    reportNumber++;
+    if (OSVR_RETURN_SUCCESS != osvrClientFreeImage(*ctx, report->state.data)) {
+        printf("Error, osvrClientFreeImage call failed.\n");
+    }
+}
+
+int main() {
+    OSVR_ClientContext ctx =
+        osvrClientInit("com.osvr.exampleclients.Imaging", 0);
+    if(!ctx) {
+        printf("Error, could not initialize client context. The server may not be running.\n");
+        return -1;
+    }
+
+    OSVR_ClientInterface camera = NULL;
+    if(OSVR_RETURN_SUCCESS != osvrClientGetInterface(ctx, "/camera", &camera)) {
+        printf("Error, could not get the camera interface at /camera.\n");
+        return -1;
+    }
+
+    /// We keep a copy of the last report to avoid de-allocating the image
+    /// buffer until we have a new report.
+    //osvr::clientkit::ImagingReportOpenCV lastReport;
+
+    // Register the imaging callback.
```

Comment:
I think we're still technically using just `/* */` for the C stuff in case it bothers old C compilers.